universe/marathon.json.mustache

```diff
@@ -46,49 +48,92 @@
-        0
+    "portDefinitions": [
+        {
+            "port": 0,
+            "protocol": "tcp",
+            "name": "dispatcher",
+            "labels": {
+                "VIP_0": "spark-dispatcher:7077"
+            }
+        },
+        {
+            "port": 0,
+            "protocol": "tcp",
+            "name": "dispatcher-ui",
+            "labels": {
+                "VIP_1": "spark-dispatcher:4040"
+            }
+        },
+        {
+            "port": 0,
+            "protocol": "tcp",
+            "name": "dispatcher-proxy",
+            "labels": {
+                "VIP_2": "spark-dispatcher:80"
+            }
+        }
     ],
+
+{{#service.UCR_containerizer}}
     "container": {
-        "type": "DOCKER",
-        "docker": {
-{{#service.docker-image}}
-            "image": "{{service.docker-image}}",
-{{/service.docker-image}}
-{{^service.docker-image}}
-            "image": "{{resource.assets.container.docker.spark_docker}}",
-{{/service.docker-image}}
-            "network": "HOST",
-{{#service.user}}
-            "parameters": [
-               {
-                 "key": "user",
-                 "value": "{{service.user}}"
-               }
-            ],
-{{/service.user}}
-            "forcePullImage": true
+    "type": "MESOS",
+    "docker": {
+    {{#service.docker-image}}
+        "image": "{{service.docker-image}}",
+    {{/service.docker-image}}
+    {{^service.docker-image}}
+        "image": "{{resource.assets.container.docker.spark_docker}}",
+    {{/service.docker-image}}
+    "forcePullImage": true
+    }
+    },
+{{/service.UCR_containerizer}}
+{{^service.UCR_containerizer}}
+    "container": {
+    "type": "DOCKER",
+    "docker": {
+    {{#service.docker-image}}
+        "image": "{{service.docker-image}}",
+    {{/service.docker-image}}
+    {{^service.docker-image}}
+        "image": "{{resource.assets.container.docker.spark_docker}}",
+    {{/service.docker-image}}
+    "network": "HOST",
+    {{#service.user}}
+        "parameters": [
+        {
+        "key": "user",
+        "value": "{{service.user}}"
         }
+        ],
+    {{/service.user}}
+    "forcePullImage": true
+    }
     },
+{{/service.UCR_containerizer}}
+
     "healthChecks": [
         {
-{{#security.ssl.enabled}}
-            "protocol": "COMMAND",
-            "command": { "value": "curl -f -k https://$HOST:$PORT2/" },
-{{/security.ssl.enabled}}
-{{^security.ssl.enabled}}
-            "path": "/",
             "portIndex": 2,
-            "protocol": "HTTP",
-{{/security.ssl.enabled}}
+            "protocol": "MESOS_HTTP",
+            "path": "/",
             "gracePeriodSeconds": 5,
             "intervalSeconds": 60,
             "timeoutSeconds": 10,
             "maxConsecutiveFailures": 3
         }
     ],
+
+    "upgradeStrategy": {
+        "minimumHealthCapacity": 0,
+        "maximumOverCapacity": 0
+    },
+
+    "unreachableStrategy": {
+        "expungeAfterSeconds": 0,
+        "inactiveAfterSeconds": 0
```

Comment:
The default value is 300 sec. Why did you change it to zero?

Reply:
We store the state in ZK, so might as well restart immediately.

Reply:
if you feel strongly I can change it back

Reply:
Makes sense. I was mistakenly thinking this applied to the drivers instead of the dispatcher.